### PR TITLE
decouple DataflowPipelineOptions #139

### PR DESCRIPTION
**Pull request by nevillelyh**

### CODE REVIEW

File: scio-core/src/main/scala/com/spotify/scio/ScioContext.scala [hunk 1]
```diff
@@ -117,23 +117,36 @@ object ScioContext {
...
+  if (options.isInstanceOf[DataflowPipelineOptions]) {
+    if (dfOptions.getAppName == null || dfOptions.getAppName.startsWith("ScioContext$")) {
+      this.setName(CallSites.getAppName)
+    }
   }
 
   /** Dataflow pipeline. */
   def pipeline: Pipeline = {
     if (_pipeline == null) {
-      options.setFilesToStage(getFilesToStage(artifacts).asJava)
```
Comment (ravwojdyla) [inline scio-core/src/main/scala/com/spotify/scio/ScioContext.scala]: `else` warning for artifacts?

**Comment by ravwojdyla** [comment]
Except the one comment - 👍

File: scio-core/src/main/scala/com/spotify/scio/ScioContext.scala (hunk 1 above)
Comment (nevillelyh) [inline scio-core/src/main/scala/com/spotify/scio/ScioContext.scala]: Right now the only other case is direct pipeline with no artifacts, so no need to warn?

Comment (ravwojdyla) [inline scio-core/src/main/scala/com/spotify/scio/ScioContext.scala]: I just don't like the silent failure - so if someone adds artifacts, and if there is more runners some time from now, will someone remember to look in this piece of code to make sure artifacts are handled.

**Pull request merged**